### PR TITLE
Adding reasoning field - tasks1-50

### PR DESCRIPTION
**Pull request by yeganehkordi**
I wrote a script for separating the reasoning field in the tasks and updated the first 50 tasks. I'll update all the files in the next PRs.

### CODE REVIEW

File: src/auto_add_reasoning.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import json
```
Comment (danyaljj) [inline src/auto_add_reasoning.py]: if we're adding the reasoning categories to the files, why do we need to check in this file?

**Review by danyaljj — APPROVED** [review]
One small suggestion; otherwise, everything looks good.

File: src/auto_add_reasoning.py (hunk 1 above)
Comment (yeganehkordi) [reply src/auto_add_reasoning.py]: Makes sense :) I'll drop it now.

**Comment by swarooprm** [comment]
All the PRs on reasoning field: Is it about separating the categories with the word 'reasoning' and storing in a different field? If this is the case, we might not need to review. @yeganehkordi

**Comment by yeganehkordi** [comment]
> All the PRs on reasoning field: Is it about separating the categories with the word 'reasoning' and storing in a different field? If this is the case, we might not need to review. @yeganehkordi

Yes, I've separated the reasoning field and sorted the keys.

**Review by swarooprm — APPROVED** [review]

**Pull request merged**